### PR TITLE
add weekly hours subtext for budgets

### LINKED ISSUE #1570: Add weekly total sub-text for budgets
**Is your feature request related to a problem? Please describe.**
It's very nice when I don't have to "click circle, click edit" to see a summary of my budget.

**Describe the solution you'd like**
Summarize the budget settings in the sub-text that appears when you first click on a budget.  
Refactor example budget at bottom to the following:  
**1 hour on weekdays  
between 14-18   
5h per week**  
  
NB: Please note the number of hours on the first and last line could also be a range of min-max.

**Describe alternatives you've considered**
None.

**Additional context**
Previouse subtext:  
image

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1570

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
The issue is resolved.  

However there seems to be a regression effect: the + is gone.

**Comment by vinaybadgujar102** [comment]
@tijlleenders It seems to work fine on my end. this could be a page-loading issue.
Screenshot from 2023-11-04 20-00-32

**Review by tijlleenders — APPROVED** [review]
Let's see what happens when we merge this...

**Comment by tijlleenders** [comment]
@Tushar-4781 OK to merge?

File: src/utils/index.ts [hunk 1]
```diff
@@ -236,3 +236,17 @@ export const getSvgForGoalPps = (count: number) => {
       return "ThreeAvatars";
   }
 };
+
+export const formatHoursText = (hours: string | null) => {
+  if (hours === null) {
+    return "";
+  }
+  if (typeof hours === "string") {
```
Comment (Tushar-4781) [inline src/utils/index.ts:244]: remove this check as you have already validated the hours variable is not null

File: src/utils/index.ts [hunk 2]
```diff
@@ -236,3 +236,17 @@ export const getSvgForGoalPps = (count: number) => {
...
+    const parts = hours.split("-").map(Number);
+    if (parts.length === 2 && parts[0] === parts[1]) {
+      return `${parts[0]} hour${parts[0] > 1 ? "s" : ""}`;
+    }
+    return `${hours} hours`;
+  }
+  return `${hours} hour${hours > 1 ? "s" : ""}`;
```
Comment (Tushar-4781) [inline src/utils/index.ts:251]: no need of this either

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
hi @vinaybadgujar102 please check these minor change

File: src/utils/index.ts (hunk 1 above)
Comment (vinaybadgujar102) [reply src/utils/index.ts:244]: `const formatHoursText = (hours: string | null) => {`
had to check for null due to the type of parameter being either string or null

Comment (Tushar-4781) [reply src/utils/index.ts]: no i m talking about   if (typeof hours === "string") {
this check

File: src/utils/index.ts [hunk 3]
```diff
@@ -236,3 +236,14 @@ export const getSvgForGoalPps = (count: number) => {
       return "ThreeAvatars";
   }
 };
+
+export const formatHoursText = (hours: string | null) => {
```
Comment (Tushar-4781) [inline src/utils/index.ts:240]: change the name of this to **formatBudgetHrsToText**

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**